src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -142,13 +142,18 @@ class CornerstoneViewport extends Component {
     const { imageIdIndex } = this.state;
     const imageId = imageIds[imageIdIndex];
 
-    // EVENTS
-    this._bindInternalEventListeners();
-    this._bindExternalEventListeners();
+    // ~~ EVENTS: CORNERSTONE
     this._handleOnElementEnabledEvent();
+    this._bindInternalCornerstoneEventListeners();
+    this._bindExternalEventListeners("cornerstone");
 
-    // Fire 'er up
+    // Enable
     cornerstone.enable(this.element, cornerstoneOptions);
+
+    // ~~ EVENTS: ELEMENT
+    this._bindInternalElementEventListeners();
+    this._bindExternalEventListeners("element");
```

Comment:
We technically need to unbind/rebind these anytime there's a disable/enable event

Reply:
This is being done in will mount/unmount already, callbacks were not being triggered when binding in every state change (will update cycle)

Reply:
That's a good point. I guess if someone is manually enabling/disabling the element inside this component, they've already broken the abstraction. No reason to guard/protect for it